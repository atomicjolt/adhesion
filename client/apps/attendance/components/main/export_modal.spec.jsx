import React from 'react';
import TestUtils from 'react-addons-test-utils';
import ExportCSV from './export_modal';

describe('Export CSV', () => {
  const props = {
    apiUrl: '',
    lmsCourseId: '123',
    downloadFile: () => {},
    onExport: () => {},
    closeModal: () => {},
  };

  it('renders', () => {
    const result = TestUtils.renderIntoDocument(<ExportCSV {...props} />);
    expect(result).toBeDefined();
  });

  it('should call onExport when export button is clicked', () => {
    const result = TestUtils.renderIntoDocument(<ExportCSV {...props} />);

    spyOn(result, 'onExport');
    const button1 = TestUtils.scryRenderedDOMComponentsWithClass(result, 'c-btn--export')[0];
    const button2 = TestUtils.scryRenderedDOMComponentsWithClass(result, 'c-btn--export')[1];

    TestUtils.Simulate.click(button1);
    TestUtils.Simulate.click(button2);

    expect(result.onExport).toHaveBeenCalledTimes(2);
  });

  it('should call onOut when someone clicks outside', () => {
    spyOn(props, 'closeModal');
    const result = TestUtils.renderIntoDocument(<ExportCSV {...props} />);
    const outside = TestUtils.findRenderedDOMComponentWithClass(result, 'c-popup--outside');
    TestUtils.Simulate.click(outside);
    expect(props.closeModal).toHaveBeenCalled();
  });
});
