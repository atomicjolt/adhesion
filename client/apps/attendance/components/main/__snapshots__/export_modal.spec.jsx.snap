// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export CSV renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
        Object {
            "height": "100%",
            "position": "absolute",
            "top": "0px",
            "width": "100%",
          }
    }
>
    <div
        className="c-popup  c-popup--export  is-open"
        role="radioGroup"
    >
        <button
            className="c-btn c-btn--cancel"
            onClick={[Function]}
        >
            <i
                className="material-icons"
            >
                clear
            </i>
        </button>
        <div
            className="c-popup__left"
        >
            <ExportButton
                ariaPosinset={1}
                onExport={[Function]}
                text="Export All"
            />
        </div>
        <div
            className="c-popup__right"
        >
            <RangePicker
                endDate={2016-01-31T07:00:00.000Z}
                onEndChange={[Function]}
                onStartChange={[Function]}
                startDate={2016-01-31T07:00:00.000Z}
            />
            <ExportButton
                ariaPosinset={2}
                downloadOptions={
                    Object {
                        "endDate": 2016-01-31T07:00:00.000Z,
                        "startDate": 2016-01-31T07:00:00.000Z,
                      }
                }
                onExport={[Function]}
                text="Export Date Range"
            />
        </div>
    </div>
    <button
        className="c-popup--outside"
        onClick={[Function]}
        style={
            Object {
                "height": "100vh",
                "opacity": "0",
                "position": "absolute",
                "top": "0px",
                "width": "100vw",
                "zIndex": 2,
              }
        }
    />
</div>,
  "nodes": Array [
    <div
      style={
            Object {
                  "height": "100%",
                  "position": "absolute",
                  "top": "0px",
                  "width": "100%",
                }
      }
>
      <div
            className="c-popup  c-popup--export  is-open"
            role="radioGroup"
      >
            <button
                  className="c-btn c-btn--cancel"
                  onClick={[Function]}
            >
                  <i
                        className="material-icons"
                  >
                        clear
                  </i>
            </button>
            <div
                  className="c-popup__left"
            >
                  <ExportButton
                        ariaPosinset={1}
                        onExport={[Function]}
                        text="Export All"
                  />
            </div>
            <div
                  className="c-popup__right"
            >
                  <RangePicker
                        endDate={2016-01-31T07:00:00.000Z}
                        onEndChange={[Function]}
                        onStartChange={[Function]}
                        startDate={2016-01-31T07:00:00.000Z}
                  />
                  <ExportButton
                        ariaPosinset={2}
                        downloadOptions={
                              Object {
                                    "endDate": 2016-01-31T07:00:00.000Z,
                                    "startDate": 2016-01-31T07:00:00.000Z,
                                  }
                        }
                        onExport={[Function]}
                        text="Export Date Range"
                  />
            </div>
      </div>
      <button
            className="c-popup--outside"
            onClick={[Function]}
            style={
                  Object {
                        "height": "100vh",
                        "opacity": "0",
                        "position": "absolute",
                        "top": "0px",
                        "width": "100vw",
                        "zIndex": 2,
                      }
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ExportModal
        apiUrl=""
        closeModal={[Function]}
        downloadFile={[Function]}
        lmsCourseId="123"
        onExport={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ExportModal {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "apiUrl": "",
          "closeModal": [Function],
          "downloadFile": [Function],
          "lmsCourseId": "123",
          "onExport": [Function],
        },
        "refs": Object {},
        "state": Object {
          "endDate": 2016-01-31T07:00:00.000Z,
          "startDate": 2016-01-31T07:00:00.000Z,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
                    Object {
                              "height": "100%",
                              "position": "absolute",
                              "top": "0px",
                              "width": "100%",
                            }
          }
>
          <div
                    className="c-popup  c-popup--export  is-open"
                    role="radioGroup"
          >
                    <button
                              className="c-btn c-btn--cancel"
                              onClick={[Function]}
                    >
                              <i
                                        className="material-icons"
                              >
                                        clear
                              </i>
                    </button>
                    <div
                              className="c-popup__left"
                    >
                              <ExportButton
                                        ariaPosinset={1}
                                        onExport={[Function]}
                                        text="Export All"
                              />
                    </div>
                    <div
                              className="c-popup__right"
                    >
                              <RangePicker
                                        endDate={2016-01-31T07:00:00.000Z}
                                        onEndChange={[Function]}
                                        onStartChange={[Function]}
                                        startDate={2016-01-31T07:00:00.000Z}
                              />
                              <ExportButton
                                        ariaPosinset={2}
                                        downloadOptions={
                                                  Object {
                                                            "endDate": 2016-01-31T07:00:00.000Z,
                                                            "startDate": 2016-01-31T07:00:00.000Z,
                                                          }
                                        }
                                        onExport={[Function]}
                                        text="Export Date Range"
                              />
                    </div>
          </div>
          <button
                    className="c-popup--outside"
                    onClick={[Function]}
                    style={
                              Object {
                                        "height": "100vh",
                                        "opacity": "0",
                                        "position": "absolute",
                                        "top": "0px",
                                        "width": "100vw",
                                        "zIndex": 2,
                                      }
                    }
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
                    Object {
                              "height": "100%",
                              "position": "absolute",
                              "top": "0px",
                              "width": "100%",
                            }
          }
>
          <div
                    className="c-popup  c-popup--export  is-open"
                    role="radioGroup"
          >
                    <button
                              className="c-btn c-btn--cancel"
                              onClick={[Function]}
                    >
                              <i
                                        className="material-icons"
                              >
                                        clear
                              </i>
                    </button>
                    <div
                              className="c-popup__left"
                    >
                              <ExportButton
                                        ariaPosinset={1}
                                        onExport={[Function]}
                                        text="Export All"
                              />
                    </div>
                    <div
                              className="c-popup__right"
                    >
                              <RangePicker
                                        endDate={2016-01-31T07:00:00.000Z}
                                        onEndChange={[Function]}
                                        onStartChange={[Function]}
                                        startDate={2016-01-31T07:00:00.000Z}
                              />
                              <ExportButton
                                        ariaPosinset={2}
                                        downloadOptions={
                                                  Object {
                                                            "endDate": 2016-01-31T07:00:00.000Z,
                                                            "startDate": 2016-01-31T07:00:00.000Z,
                                                          }
                                        }
                                        onExport={[Function]}
                                        text="Export Date Range"
                              />
                    </div>
          </div>
          <button
                    className="c-popup--outside"
                    onClick={[Function]}
                    style={
                              Object {
                                        "height": "100vh",
                                        "opacity": "0",
                                        "position": "absolute",
                                        "top": "0px",
                                        "width": "100vw",
                                        "zIndex": 2,
                                      }
                    }
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ExportModal
    apiUrl=""
    closeModal={[Function]}
    downloadFile={[Function]}
    lmsCourseId="123"
    onExport={[Function]}
/>,
}
`;
